test/end-to-end.js

```diff
@@ -1185,3 +1189,9 @@ function readBuiltinPhasesFromSanbox() {
   app.lazyrouter(); // initialize request handling phases
   return app._requestHandlingPhases;
 }
+
+function getSandBoxApp() {
+  var app = require(SANDBOX);
+  app.use(app.get('restApiRoot'), app.loopback.rest());
```

Comment:
I am afraid this is removing an important point of these end-to-end tests, which is to verify the behaviour of the application scaffolded by loopback-workspace. Also note that this is registering the middleware after all other middleware from `server/middleware.json` has been already registered. The order in which the middleware is registered is important, as Express checks the middleware one by one until some handler matches. If the app was scaffolded and bootstrapped correctly, then this line would be a no-op because all loopback applications add `urlNotFound` handler that converts all requests to a 404 error.

I suspect that `require(SANDBOX)` is not correctly booting up the application.

Most likely, the bootstrapping process is asynchronous and you need to wait for `app.once('booted')` before continuing with the test.